aggregation-worker/src/main/java/au/org/emii/geoserver/client/HttpIndexReader.java

```diff
@@ -137,6 +140,78 @@ public HttpIndexReader(String geoserver) {
     }
 
 
+    public String getLatestTimeStep(String layer, String timeField) throws AggregationException {
+
+        String geoserverWfsEndpoint = String.format("%s/wfs", geoserver);
```

Comment:
Would it be possible to pull out the common code here to prevent divergence - this is all pretty much the same as the above apart from the time clause.

Reply:
Refactored as much as possible.